src/Entities/Player.cpp

```diff
@@ -2643,6 +2643,64 @@ void cPlayer::SendBlocksAround(int a_BlockX, int a_BlockY, int a_BlockZ, int a_R
 
 bool cPlayer::PlaceBlocks(const sSetBlockVector & a_Blocks)
 {
+	// Check to see if any entity intersects any block being placed
+	class DoesIntersectBlock : public cEntityCallback
+	{
+	public:
+		const sSetBlockVector & m_Blocks;
+
+		// The distance inside the block the entity can still be.
+		const double EPSILON = 0.05;
+
+		DoesIntersectBlock(const sSetBlockVector & a_Blocks) :
+			m_Blocks(a_Blocks)
+		{
+		}
+
+		virtual bool Item(cEntity * a_Entity) override
+		{
+			cBoundingBox EntBox(a_Entity->GetPosition(), a_Entity->GetWidth() / 2, a_Entity->GetHeight());
+			for (auto blk: m_Blocks)
+			{
+				cBoundingBox BlockBox(
+					blk.GetX() + EPSILON,
+					blk.GetX() + 1.0 - EPSILON,
+					blk.GetY() + EPSILON,
+					blk.GetY() + 1.0 - EPSILON,
+					blk.GetZ() + EPSILON,
+					blk.GetZ() + 1.0 - EPSILON
+				);
+				if (EntBox.DoesIntersect(BlockBox))
+				{
+					return true;
+				}
+			}
+			return false;
+		}
+	} Callback(a_Blocks);
+
+	// Compute the bounding box of the blocks to be placed
+	double MaxDouble = std::numeric_limits<double>::max();
+	double MinDouble = std::numeric_limits<double>::min();
+	Vector3d BlocksMin(MaxDouble, MaxDouble, MaxDouble);
+	Vector3d BlocksMax(MinDouble, MinDouble, MinDouble);
+	for (auto blk: a_Blocks)
+	{
+		BlocksMin = BlocksMin.Min(blk.GetPos());
+		BlocksMax = BlocksMax.Max(blk.GetPos() + Vector3d(1, 1, 1));
+	}
```

Comment:
I can't think of any blocks that depend on their surroundings? I think all blocks collision boxes can be determined from the type and the block meta (I'm a little fuzzy on how all that works, but fences store the direction they go in as part of the block).

And the collision box may not even be a cube, in the case of stairs.